File: olzdata/pages/admin_clawback.pl
print <<EOS;

<link rel="stylesheet" href="$commission_engine_api_url/css/app.css?v=$app_css_version" />
<link rel="stylesheet" href="https://cdn.datatables.net/responsive/2.2.3/css/responsive.bootstrap.min.css">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/eonasdan-bootstrap-datetimepicker/4.17.47/css/bootstrap-datetimepicker.min.css" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.7/css/select2.min.css" />
<link rel="stylesheet" href="$commission_engine_api_url/css/datepicker.css?v=$app_css_version" />
<link rel="stylesheet" href="$commission_engine_api_url/css/select2-bootstrap.css"/>
<link rel="stylesheet" href="$commission_engine_api_url/css/admin_clawback.css?v=1" />

<div class="tool-container tool-container--default" id="app-clawback">

    <div class="row">
        <div class="col-md-12">
            <h4>CLAWBACK/REFUND/MOVE ORDERS</h4>
            <hr />
        </div>
    </div>
    <div class="row">
        <div class="col-lg-12" >
            <ul class="nav nav-tabs" role="tablist" id='nav-tab-report'>
                <li class="nav-item">
                    <a class="nav-link active" href="#tab-orders" aria-controls="tab-orders" role="tab" data-toggle="tab" style="color: black !important;">Orders</a>
                </li>
                <li class="nav-item">
                    <a class="nav-link" href="#tab-logs" aria-controls="tab-logs" role="tab" data-toggle="tab" style="color: black !important;">Move Logs</a>
                </li>
            </ul>
            <!-- Tab panes -->
            <div class="tab-content" style="padding: 15px;border: 1px solid #ddd;border-top: none;">
                <div role="tabpanel" class="tab-pane active" id="tab-orders">
                    <div class="row mb-5">
                        <div class="col-md-4">
                            <form class="form-horizontal">
                                <div class="form-group">
                                    <div class="col-md-12">
                                        <label for="start-date">From</label>
                                        <input id="start-date" type="text" class="form-control flat">
                                    </div>
                                </div>
                                <div class="form-group">
                                    <div class="col-md-12">
                                        <label for="end-date">To</label>
                                        <input id="end-date" type="text" class="form-control flat">
                                    </div>
                                </div>
                                <div class="form-group">
                                    <div class="col-md-12">
                                        <label for="purchaser-id">Purchaser (Optional)</label>
                                        <select2-autocomplete-member id="purchaser-id" :url="autocompleteUrl" v-model="filters.memberId"></select2-autocomplete-member>
                                    </div>
                                </div>
                                <div class="form-group">
                                    <div class="col-sm-12">
                                        <button id="button-view-orders" type="button" class="btn btn-primary btn-block" >View Orders</button>
                                    </div>
                                </div>
                            </form>
                        </div>
                    </div>

                    <table id="table-orders" class="table table-striped table-bordered dt-responsive nowrap" style="width:100%;">
                        <thead class="table__header table__header--bg-primary">
                        <tr class="table__row">
                            <th class="table__cell">Order ID</th>
                            <th class="table__cell">Invoice</th>
                            <th class="table__cell">Purchaser</th>
                            <th class="table__cell">Desc</th>
                            <th class="table__cell">Date Purchased</th>
                            <th class="table__cell">CV</th>
                            <th class="table__cell">Amount</th>
                            <th class="table__cell">Set By</th>
                            <th class="table__cell">Action</th>
                        </tr>
                        </thead>
                        <tbody class="table__body">
                        </tbody>

                    </table>
                </div>
                <div role="tabpanel" class="tab-pane" id="tab-logs">

                    <table id="table-logs" class="table table-bordered" style="width: 100%;">
                        <thead class="table__header table__header--bg-primary">
                        <tr class="table__row">
                            <th class="table__cell">Order ID</th>
                            <th class="table__cell">New Purchaser</th>
                            <th class="table__cell">Old Purchaser</th>
                            <th class="table__cell">New Transaction Date</th>
                            <th class="table__cell">Old Transaction Date</th>
                            <th class="table__cell">Date</th>
                            <th class="table__cell">Changed by</th>
                        </tr>
                        </thead>
                        <tbody class="table__body">
                        </tbody>
                    </table>

                </div>
            </div>

        </div>
    </div>

    <div class="modal fade" id="modal-order-items" role="dialog" aria-labelledby="modal-order-items-label">
        <div class="modal-dialog modal-xl modal-lg" role="document">
            <form class="modal-content" v-on:submit.prevent>
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-order-items-label">ORDER ID {{ transaction_id }} - ORDER ITEMS</h4>
                </div>
                <div class="modal-body">

                    <div class="table-responsive">
                        <table id="table-order-items" class="table table table-striped" >
                            <!--Table head-->
                            <thead class="table__header table__header--bg-primary">
                            <tr class="table__row">
                                <th class="table__cell col-md-2">ID</th>
                                <th class="table__cell col-md-3 text-left">Name</th>
                                <th class="table__cell col-md-1">Qty</th>
                                <th class="table__cell col-md-1">CV</th>
                                <th class="table__cell col-md-1">Total CV</th>
                                <th class="table__cell col-md-1">Refunded Qty</th>
                                <th class="table__cell col-md-2">Total CV Refunded</th>
                                <th class="table__cell col-md-1">Refund Qty</th>
                            </tr>
                            </thead>
                            <!--Table head-->
                            <tbody class="table__body">
                            <tr v-if="products.length == 0" class="table__row text-center">
                                <td colspan="6">Fetching <i class="fa fa-spinner fa-pulse"></i></td>
                            </tr>
                            <tr v-for="(product, index) in products" class="table__row">
                                <td class="table__cell text-center">{{ product.transaction_product_id }}</td>
                                <td class="table__cell text-left">
                                    {{ product.name }}
                                    <ul  style="padding-left: 20px;font-size: 12px;">
                                        <li class="text-danger" v-for="(option, i) in product.options">{{ option.name }}: {{ option.value }}</li>
                                    </ul>
                                </td>
                                <td class="table__cell text-center">{{ product.quantity }}</td>
                                <td class="table__cell text-right">{{ product.price }}</td>
                                <td class="table__cell text-right">{{ product.total }}</td>
                                <td class="table__cell text-center">{{ product.refunded_quantity }}</td>
                                <td class="table__cell text-right">{{ product.refunded_amount }}</td>
                                <td class="table__cell text-center">
                                    <input :name="p(product.transaction_product_id, 'order_id')" v-model="product.order_id" type="hidden">
                                    <input :name="p(product.transaction_product_id, 'transaction_product_id')" v-model="product.transaction_product_id" type="hidden">
                                    <input :readonly="readonlyProduct(order) == 1 || product.quantity < 1" :name="p(product.transaction_product_id, 'refund_quantity')" v-model="product.refund_quantity" type="text" class="form-control input-sm">
                                </td>
                            </tr>
                            </tbody>
                        </table>
                    </div>
                    <div id="form-refund-order-products-error" class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>
                </div>
                <div class="modal-footer">
                    <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
                    <button v-on:click.prevent="onSubmit('merchant')" type="submit" class="btn btn-primary hidden" id="btn-save-refund-products">Refund</button>
                    <!--<button v-on:click.prevent="onSubmit('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-products-commission">Refund (Commission Only)</button>-->
                    <button v-on:click.prevent="onSubmit('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-products-commission">Refund</button>
                </div>
            </form>
        </div>
    </div>

    <div class="modal fade" id="modal-refund-order" role="dialog" aria-labelledby="modal-refund-order-label">
        <div class="modal-dialog modal-lg" role="document">
            <form class="modal-content form-horizontal" id="form-refund-order" v-on:submit.prevent>
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                    <h4 class="modal-title" id="modal-refund-order-label">CLAWBACK/REFUND ORDER ID {{ order.order_id }}</h4>
                </div>
                <div class="modal-body">
                    <table class="table table table-striped" style="width:100%">
                        <tbody>
                        <tr>
                            <th>ORDER ID</th>
                            <td><span class="r-order-id">{{ order.order_id }}</span></td>
                        </tr>
                        <tr>
                            <th>PURCHASER</th>
                            <td><span class="r-purchaser">{{ order.purchaser }}</span></td>
                        </tr>
                        <tr>
                            <th>SUB-TOTAL</th>
                            <td class="text-right">\$<span class="r-sub-total">{{ order.sub_total }}</span></td>
                        </tr>
                        <tr>
                            <th>TAX</th>
                            <td class="text-right">\$<span class="r-tax">{{ order.tax }}</span></td>
                        </tr>
                        <tr>
                            <th>SHIPPING</th>
                            <td class="text-right">\$<span class="r-shipping">{{ order.shipping_fee }}</span></td>
                        </tr>
                        <tr>
                            <th>TOTAL</th>
                            <td class="text-right">\$<span class="r-total">{{ order.total }}</span></td>
                        </tr>
                        <tr>
                            <th>CV</th>
                            <td class="text-right">\${{ formatMoney(order.commission_value) }}</td>
                        </tr>
                        </tbody>
                    </table>

                    <input type="hidden" name="transaction_id" v-model="order.order_id"/>
                    <div class="form-group hidden">
                        <div class="col-sm-12">
                            <label for="order_percentage">% Off</label>
                            <input :readonly="order.is_clawback ? true : (order.is_full_order ? true : false)" v-on:keyup="keyupPercent" v-model="order.percentage_off" type="text" id="order_percentage" name="percent" class="form-control">
                        </div>
                    </div>
                    <div class="form-group hidden">
                        <div class="col-sm-12 text-center">
                            <p class="h4" style="margin: 0;">-- OR --</p>
                        </div>
                    </div>
                    <div class="form-group hidden">
                        <div class="col-sm-12">
                            <label for="order_amount">Amount Off</label>
                            <input :readonly="order.is_clawback ? true : (order.is_full_order ? true : false)" v-on:keyup="keyupAmount" v-model="order.amount_off" type="text" id="order_amount" name="amount" class="form-control">
                        </div>
                    </div>
                    <div class="form-group hidden">
                        <div class="col-sm-12 text-center">
                            <p class="h4" style="margin: 0;">-- OR --</p>
                        </div>
                    </div>
                    <div class="checkbox hidden">
                        <label><input :disabled="order.is_clawback ? true : false" :readonly="order.is_clawback ? true : false" v-on:change="changeIsFullOrder" v-model="order.is_full_order" type="checkbox" value="1" name="is_full_order">Full Order</label>
                    </div>
                    <div id="form-refund-order-error" class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>

                </div>
                <div class="modal-footer">
                    <a type="button" class="btn btn-default" data-dismiss="modal">Close</a>
                    <button v-if="!order.is_clawback" v-on:click.prevent="onSubmitRefund('merchant')" type="submit" class="btn btn-primary hidden" id="btn-save-refund">Refund</button>
                    <!--<button v-if="!order.is_clawback" v-on:click.prevent="onSubmitRefund('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-commission">Refund (Commission Only)</button>-->
                    <button v-if="!order.is_clawback" v-on:click.prevent="onSubmitRefund('commission')" type="submit" class="btn btn-primary" id="btn-save-refund-commission">Refund</button>
                </div>
            </form>
        </div>
    </div>

    <!-- Move Order Modal -->
    <div class="modal fade" id="modal-move" role="dialog" aria-labelledby="myModalLabel">
        <div class="modal-dialog" role="document">
            <form class="modal-content form-horizontal" id="form-move" v-on:submit.prevent="save">
                <div class="modal-header">

                    <h4 class="modal-title" id="myModalLabel">ORDER ID {{ order.order_id }}</h4>
                </div>
                <div class="modal-body">
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="invoice">Invoice</label>
                            <input class="form-control" id="invoice" v-model="order.invoice" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="transaction-date">Transaction Date</label>
                            <input class="form-control" id="transaction-date" v-model="order.transaction_date" onkeydown="return false"/>
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="purchaser">Purchaser</label>
                            <input class="form-control" id="purchaser" v-model="displayPurchaser" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="sponsor">Sponsor</label>
                            <input class="form-control" id="sponsor" v-model="displaySponsor" readonly />
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <label for="new-purchaser-id">New Purchaser</label><br/>
                            <select2-autocomplete-member id="new-purchaser-id" :url="autocompleteUrl" v-model="order.new_purchaser_id"></select2-autocomplete-member>
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12">
                            <div class="form-group form-check">
                                <input type="checkbox" :name="" class="form-check-input" id="">
                                <label class="form-check-label" for="exampleCheck1">Mark as Sharing link order</label>
                            </div>
                        </div>
                    </div>
                    <div class="form-group" v-show="error.message != null">
                        <div class="col-sm-12 has-error text-center">
                            <span class="help-block">{{ error.message }}</span>
                        </div>
                    </div>

                </div>
                <div class="modal-footer">
                    <a type="button" class="btn btn-default" data-dismiss="modal">Close</a>
                    <button type="submit" class="btn btn-primary" id="btn-save" >Update</button>
                </div>
            </form>
        </div>
    </div>

</div>

<script src="$commission_engine_api_url/js/app.js?v=$app_js_version"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.7/js/select2.full.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/dataTables.responsive.min.js"></script>
<script src="https://cdn.datatables.net/responsive/2.2.3/js/responsive.bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.8.0/js/bootstrap-datepicker.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/eonasdan-bootstrap-datetimepicker/4.17.47/js/bootstrap-datetimepicker.min.js"></script>
<script src="$commission_engine_api_url/js/admin_clawback.js?v=1"></script>

EOS
1;

File: 515data/public/js/admin_clawback.js
(function ($, api_url, Vue, swal, axios, location, moment, undefined) {

    const client = commissionEngine.createAccessClient();
    commissionEngine.setupAccessTokenJQueryAjax();

    $.fn.ddatepicker = $.fn.datepicker; // jquery-ui is overriding the bootstrap-datepicker

    let $dtLogs;
    let $dt;
	let $payouts;
	
	let boxes_checked = 0;

    const vm = new Vue({
        el: "#app-clawback",
        data: function () {
            return {

                autocompleteUrl: `${api_url}common/autocomplete/members`,
                autocompletePodUrl: `${api_url}common/autocomplete/pods`,

                filters: {
                    startDate: null,
                    endDate: null,
                    memberId: null,
                    pod: null,
                },
                products: [],
                transaction_id: null,
                is_clawback: 0,

                order: {
                    order_id: null,
                    purchaser_id: null,
                    purchaser: null,
                    sponsor_id: null,
                    sponsor: null,
                    sub_total: null,
                    transaction_date: null,
                    tax: null,
                    shipping_fee: null,
                    total: null,
                    percentage_off: null,
                    amount_off: null,
                    commission_value: null,
                    is_full_order: null,
                    new_purchaser_id: null,
                    is_clawback: 0
                },

                error: {
                    message : null,
                },

                is_saving: 0,


				dtPayouts: null,
				
				boxes_checked: 0
            };
        },
        mounted() {
            this.initializeDataTables();
            this.initializeDatePicker();
            this.initializeJQueryEvents();
        },
        methods: {
            initializeDataTables() {
                let _this = this;

                $dt = $("#table-orders").DataTable({
                    processing: true,
                    serverSide: true,
					responsive: true,
                    pageLength: 25,
					/*
                     ajax: {
                         "url": api_url + 'admin/clawback/pea',
                         "data": function (d) {
                             d.startDate = $("#start-date").val();
                             d.endDate = $("#end-date").val();
                             d.memberId = _this.filters.memberId;
                         },
					},
					*/
                    ajax: {
						"url": api_url + 'admin/clawback/pea',
						"data": function (d) {
							d.startDate = $("#start-date").val();
							d.endDate = $("#end-date").val();
							d.memberId = _this.filters.memberId;
							d.pod = _this.filters.pod;
						},
                    },
                    columns: [
                        {data: 'reference_id'},
                        {data: 'associate_id'},
                        {data: 'associate_name'},
                        {data: 'customer_name'},
                        {data: 'account_type'},
                        {data: 'date_accepted'},
                        {data: 'date_started_flowing'},
                        {data: 'status'},
                        {
                            width: '200px',
                            render: function (data, type, row, meta) {
                                return '<div class="btn-group-xs" role="group" aria-label="...">' +
                                    '<button ' + ((+row.is_clawback && +row.is_per_product) ? 'disabled' : '') + ' type="button" class="btn btn-danger btn-order-refund" data-userid="'+row.transaction_id+'" style="margin-right: 5px;">Clawback</button>' +
                                    '</div>';
                            }
                        },
                    ],
                    columnDefs: [

                        {responsivePriority: 1, targets: 0},
                        {responsivePriority: 2, targets: -1},
					],
					drawCallback: function( settings ) {
						$('.btn-order-refund').on('click', function () {
							$payouts.ajax.url(api_url + 'admin/clawback/payouts?user_id='+$(this).data('userid')+'&startDate='+$("#start-date").val()+'&endDate='+$("#end-date").val()).load(); 
							$('#modal-refund-order').modal('show');
							//_this.showOrder(data);
						});
					}
				});
				
				$payouts = $("#user-payouts").DataTable({
					ajax: {
						"url": api_url + 'admin/clawback/payouts?user_id=&startDate=&endDate=',
					   },
					columns     : [
						/*
						{
							data    : 'associate',
							width: '33px',
							sortable: false,
							render: function ( data, type, row ) {
								return '<div class="custom-control custom-checkbox">'+
									'<input type="checkbox" class="custom-control-input" item-id="'+data+'" data-userid="'+data+'">'+
									'<span class="custom-control-indicator"></span>'+
									'</div>';
							}
						},
						*/
						{data    : 'associate'},
						{data    : 'associate_name'},
						{data    : 'commission_type'},
						{data    : 'commission_period'},
						{data    : 'commission_value'},
						{data    : 'amount'},
						{data    : 'date_clawed'},
						{
							data    : 'amount',
							render: function (data, type, row, meta) {
								return '<input disabled type="number" id="row-'+row.commission_payout_id+'-amount" value="'+row.commission_value+'">';
							}
						},
						{
							render: function (data, type, row, meta) {
								return '<button type="button" class="btn btn-primary btn-refund" data-id="'+row.commission_payout_id+'" data-transactionid="'+row.transaction_id+'">Clawback</button>';
							}
						}
					],
					drawCallback: function( settings ) {
						$('#user-payouts td').on('click', '.custom-checkbox', function (event) {
							var checkbox = $(this).find('.custom-control-input');
							if (checkbox.is(":checked")) {
								checkbox.prop('checked', true);
								boxes_checked++;
							} else {
								checkbox.prop('checked', false);
								boxes_checked--;
							}
							if (boxes_checked === 1) {
								$('#btn-save-refund-commission').removeClass('disabled');
							} else if (boxes_checked === 2 || boxes_checked > 2) {
								$('#btn-save-refund-commission').removeClass('disabled');
							} else {
								$('#btn-save-refund-commission').addClass('disabled');
							}
						});

						$('.btn-refund').on('click', function () {
							_this.error.message = '';
							_this.onSubmitRefund($(this).attr('data-id'), $(this).attr('data-transactionid'));
						});
					}
				});
            },
            initializeDatePicker() {
                let _this = this;

                $('#start-date').ddatepicker({
                    "setDate": new Date(),
                    "format": "yyyy-mm-dd"
                }).on('changeDate', function (e) {
                    $('#end-date').ddatepicker('setStartDate', e.date);

                    if ($('#end-date').ddatepicker('getDate') < e.date) {
                        $('#end-date').ddatepicker('setDate', e.date);
                    }
                });

                $('#end-date').ddatepicker({
                    "setDate": new Date(),
                    "startDate": new Date(),
                    "format": "yyyy-mm-dd"
                });

                $('#start-date').ddatepicker('setDate', new Date());
                $('#end-date').ddatepicker('setDate', new Date());

                // Transaction Date
                $('#transaction-date').datetimepicker({
                    defaultDate: moment(),
                    format: 'YYYY-MM-DD HH:mm:ss',
                });

                $('#transaction-date').on('dp.change', e => {_this.order.transaction_date = $('#transaction-date').val()});
            },
            initializeJQueryEvents() {

                let _this = this;

                $('#button-view-orders').on('click', function () {
                    $dt.clear().draw();
                    $dt.responsive.recalc();
                });

                $('#table-orders tbody').on('click', '.btn-view-items', function () {
                    let data = $dt.row($(this).parents('tr')).data();
                    _this.getProducts(data.order_id, data.is_clawback, data);
                });

                $('#table-orders tbody').on('click', '.btn-order-move', function () {
                    let data = $dt.row($(this).parents('tr')).data();
                    _this.showMoveOrder(data);
				});
            },
            readonlyProduct: function(order) {
                return +order.is_clawback && !+order.is_per_product;
            },
            formatMoney: function(value) {
                if(value == null || value == undefined) return 0.00;

                return parseFloat(value).toFixed(2);
            },
            changeIsFullOrder: function(event) {
                if(this.order.is_full_order) {
                    this.order.amount_off = 0;
                    this.order.percentage_off = 0;
                }
            },
            keyupAmount: function(event) {
                this.order.percentage_off = null;
            },
            keyupPercent: function(event) {
                this.order.amount_off = null;
            },
            showOrder: function(data) {
                this.error.message = null;

                this.order = JSON.parse(JSON.stringify(data));
                this.order.amount_off = data.amount_off ? data.amount_off : data.commission_value;
                // this.order.is_full_order = data.is_full_order;
                $('#modal-refund-order').modal('show');
            },
            getProducts: function (transaction_id, is_clawback, order) {

                this.order = JSON.parse(JSON.stringify(order));
                this.transaction_id = transaction_id;

                client.get(`admin/clawback/order-products/${transaction_id}`).then(response => {
                    this.products = response.data;
                    this.is_clawback = is_clawback;
                    this.error.message = null;
                    $('#modal-order-items').modal('show');
                }).catch(error => {
                    swal('Unable to fetch!','','error');
                });
            },

            p: function (index, name) {
                return 'products[' + index + '].' + name;
            },
            onSubmit: function (type) {

                // if(this.is_clawback) return;

                let _this = this;
                swal({
                    title: "Are you sure you want to clawback/refund Order ID " + this.transaction_id + " - Items? " + (type == 'commission' ? "(Commission Only)" :"(Merchant & Commission)"),
                    text: "You cannot undo this.",
                    type: "warning",
                    showCancelButton: true,
                    confirmButtonClass: "btn-danger",
                    confirmButtonText: "Confirm Refund",
                    closeOnConfirm: false,
                    showLoaderOnConfirm: true,
                }, function () {

                    client.post(`admin/clawback/refund-order-products`, {
                        products : _this.products,
                        transaction_id: _this.transaction_id,
                        set_user_id: $('#member').val(),
                        type: type
                    }).then(response => {

                        $dt.clear().draw();
                        $dt.responsive.recalc();

                        _this.products = [];
                        _this.transaction_id = null;

                        $('#modal-order-items').modal('hide');
                        swal("Successfully added for clawback/refund!", "", "success");
                    }).catch(this.axiosErrorHandler).finally(()=> {

                    });

                });
            },
            onSubmitRefund: function (commission_payout_id, transaction_id){

                var $this = this;

                swal({
                        title: "Are you sure you want to clawback/refund Payout ID " + commission_payout_id + "?",
                        text: "You cannot undo this.",
                        type: "warning",
                        showCancelButton: true,
                        confirmButtonClass: "btn-danger",
                        confirmButtonText: "Confirm Refund",
                        closeOnConfirm: false,
                        showLoaderOnConfirm: true,
                    },
                    function () {

						$.post(api_url + 'admin/clawback/refund-order', {
							payout_id : commission_payout_id,
							amount: $('#row-' + commission_payout_id + '-amount').val(),
							type: 'commission',
							set_user_id: $('#member').val(),
							transaction_id: transaction_id
						}, function (result) {
                            console.log(result);
                            $('#modal-refund-order').modal('hide');
                            swal("Successfully added for clawback/refund!", "", "success");
                        }, 'json').fail(function (xhr, status, error) {
							console.log(xhr.responseJSON);
                            if(xhr.responseJSON.message != undefined && xhr.responseJSON.message == "Validation error") {
                                let errors = xhr.responseJSON.errors;
                                $this.error.message = errors[Object.keys(errors)[0]][0];
                            } else if(xhr.responseJSON.error != undefined && typeof xhr.responseJSON.error === "string") {
                                $this.error.message = xhr.responseJSON.error;
                            } else if(xhr.responseJSON.message != undefined) {
                                $this.error.message = xhr.responseJSON.message;
                            } else {
                                $this.error.message = xhr.responseJSON.error.message;
                            }
                            swal.close();
						});

                    });
            },
            showMoveOrder(order) {
                this.order.invoice = order.invoice;
                this.order.order_id = order.order_id;
                this.order.purchaser = order.purchaser;
                this.order.purchaser_id = order.purchaser_id;
                this.order.sponsor = order.sponsor;
                this.order.sponsor_id = order.sponsor_id;
                this.order.transaction_date = order.transaction_date;
                this.order.new_purchaser_id = null;
                this.error.message = null;

                $('#modal-move').modal({backdrop: 'static', keyboard: false});
            },
            save() {
                swal({
                    title: `Are you sure you want to update Order ID ${this.order.order_id}?`,
                    // text: "You cannot undo this.",
                    type: "warning",
                    confirmButtonClass: "btn-success",
                    confirmButtonText: "Confirm",
                    cancelButtonText: "Cancel",
                    showCancelButton: true,
                    closeOnConfirm: false,
                    showLoaderOnConfirm: true,
                }, () => {

                    this.is_saving = 1;

                    client.post(`admin/move-order/orders/${this.order.order_id}/change`, {modified: $('#member').val(), ...this.order}).then(response => {

                        $('#modal-move').modal('hide');
                        // swal.close();
                        swal("Successfully updated!", "", "success");
                        $dt.clear().draw();
                        $dt.responsive.recalc();
                        $dtLogs.clear().draw();
                        $dtLogs.responsive.recalc();

                    }).catch(this.axiosErrorHandler).finally(()=> {

                        this.is_saving = 0;
                    });

                });
            },
            axiosErrorHandler(error) {

                let data = commissionEngine.parseAxiosErrorData(error.response.data);

                this.error.message = data.message;
                this.error.type = data.type;
                this.error.data = data.data;

                swal(this.error.message, "", "error");
            },
        },
        computed: {
            displayPurchaser() {
                return this.order.purchaser_id + ": " + this.order.purchaser;
            },
            displaySponsor() {
                return this.order.sponsor_id + ": " + this.order.sponsor;
            }
        }
    });


}(jQuery, window.commissionEngine.API_URL, Vue, swal, axios, window.location, moment));